test/boost/row_cache_test.cc

```diff
@@ -4219,3 +4219,58 @@ SEASTAR_TEST_CASE(test_reading_of_nonfull_keys) {
                     .produces(m1);
         });
 }
+
+SEASTAR_TEST_CASE(test_populating_cache_with_expired_and_nonexpired_tombstones) {
+    return do_with_cql_env_thread([](cql_test_env& env) {
+        replica::database& db = env.local_db();
+        auto& mm = env.migration_manager().local();
+        tests::reader_concurrency_semaphore_wrapper semaphore;
+
+        sstring ks_name = "ks";
+        sstring table_name = "table_name";
+
+        schema_ptr s = schema_builder(ks_name, table_name)
+                .with_column("pk", int32_type, column_kind::partition_key)
+                .with_column("ck", int32_type, column_kind::clustering_key)
+                .build();
+        mm.announce(
+            mm.prepare_new_column_family_announcement(s, api::new_timestamp()).get(),
+            mm.start_group0_operation().get()
+        ).get();
+
+        replica::table& t = db.find_column_family(ks_name, table_name);
+
+        auto pkey = dht::decorate_key(*s, partition_key::from_single_value(*s, serialized(1)));
```

Comment:
This will fail if there are multiple shards and the local shards happens to not own this key.
You can generate a local key like this:
https://github.com/scylladb/scylladb/blob/54170749b853e166223f865621732be4b527ffe0/test/boost/reader_concurrency_semaphore_test.cc#L1347-L1351
We also have a library method for `text` keys:
https://github.com/scylladb/scylladb/blob/54170749b853e166223f865621732be4b527ffe0/test/lib/simple_schema.hh#L42

Reply:
Thanks for comments. I used `row_cache_is_populated_using_compacting_sstable_reader` as example. Most tests in this module use the same way to get partition key as well - do you think we really have to use way for multiple shards in this test?

Reply:
I see. Done.
I've also researched test failure in debug mode described above - it was caused by incorrect using of schema - fixed

Reply:
guys, could you please check result of tests? I wonder why they failed again, tested locally in all configurations

Reply:
seems all tests passed fine finally